### PR TITLE
Don't attempt to link curses.lib on Windows

### PR DESCRIPTION
**Pull request by hughbe**
This is not available on Windows. Anyway, we don't even need this, since it is unix specific

### CODE REVIEW

File: products/llbuild/CMakeLists.txt [hunk 1]
```diff
@@ -8,6 +8,8 @@ target_link_libraries(llbuild
   llbuildCore
   llbuildBasic
   llvmSupport
-  sqlite3
-  curses
-  )
+  sqlite3)
+
+if(NOT ${CMAKE_SYSTEM_NAME} STREQUAL "Windows")
+  target_link_Libraries(curses)
```
Comment (ddunbar) [inline products/llbuild/CMakeLists.txt]: Can you fix casing here and below to match existing code (`target_link_libraries`)?

Comment (hughbe) [inline products/llbuild/CMakeLists.txt]: Whoops - fixed, thanks

**Comment by ddunbar** [comment]
@swift-ci please test

**Comment by ddunbar** [comment]
@hughbe these failures look related... something about the check is wrong I think.

**Comment by hughbe** [comment]
Whoops - this should be fixed, I've tested it on my linux VM

**Comment by ddunbar** [comment]
@swift-ci please test

**Comment by hughbe** [comment]
Not sure the tests got successfully triggered? "Waiting for status to be reported" is the status

**Comment by ddunbar** [comment]
@swift-ci please test

**Comment by ddunbar** [comment]
Ugh, apparently not -- our bot sometimes misses comments. :(

**Comment by hughbe** [comment]
I don't think the OSX failures are related

**Comment by aciidb0mb3r** [comment]
@swift-ci please test OS X

**Comment by ddunbar** [comment]
Agree OS X failures look unrelated, I tested locally.

**Pull request merged**